STACpopulator/stac_utils.py

```diff
@@ -123,6 +126,35 @@ def numpy_to_python_datatypes(data: MutableMapping[str, Any]) -> MutableMapping[
     return data
 
 
+def np2py(data):
+    """Convert numpy datatypes to python standard datatypes.
+
+    This is useful when validating against a JSON schema that does not recognize an int32 as an integer.
+
+    Parameters
+    ----------
+    data : dict, list, tuple, int, float, np.integer, np.floating, str
+      Object to convert.
+    """
+    if isinstance(data, dict):
+        return {key: np2py(value) for key, value in data.items()}
+
+    elif isinstance(data, (list, tuple)):
+        out = [np2py(item) for item in data]
+        if isinstance(data, tuple):
+            return tuple(out)
+        return out
+
+    elif issubclass(type(data), np.integer):
+        return int(data)
+
+    elif issubclass(type(data), np.floating):
+        return float(data)
+
```

Comment:
You can add an additional case here to handle the `nan` problem (see https://github.com/crim-ca/stac-populator/pull/64#discussion_r1978019325)

```python
elif isinstance(data, float) and np.isnan(data):
    return None # or "NaN" or however you think it's best to represent it
```